src/gps_to_pose_conversion_node.cpp

```diff
@@ -34,6 +35,9 @@ double g_covariance_orientation_x;
 double g_covariance_orientation_y;
 double g_covariance_orientation_z;
 std::string g_frame_id;
+std::string g_tf_child_frame_id;
+
+tf::TransformBroadcaster *tf_broadcaster;
```

Comment:
I agree with you: I hate global variables, but again, I didn't want to refactor the whole file :D
Anyway, I think a shared pointer is a good trade-off ... thanks for your valuable suggestions :)